src/consensus.rs

```diff
@@ -462,4 +465,26 @@ mod tests {
             SlotReady::Ready(_)
         ));
     }
+
+    #[tokio::test]
+    async fn wait_for_first_slot_parent_already_ready() {
+        let blockstore: Box<dyn Blockstore + Send + Sync> = Box::new(MockBlockstore::new());
+        let blockstore = Arc::new(RwLock::new(blockstore));
+
+        let slot = Slot::windows().skip(10).next().unwrap();
+        let parent = (slot.prev(), Hash::default());
```

Comment:
should this not be this?
```suggestion
        let parent = (Slot::genesis(), Hash::default());
```

Reply:
We can pick whatever parent we want here really.  The result we get back in line 486 will be the same that we provide in line 480.  The actual implementation of the Pool doesn't matter.

Reply:
ah, ok... so the `returning()` call specifies what `wait_for_parent_read()` should return when called with `slot` later?